objects/client.c

```diff
@@ -1962,21 +1953,28 @@ client_set_maximized_common(lua_State *L, int cidx, bool s, const char* type, co
 
     /* Store the current and next state on 2 bit */
     const client_maximized_t current = (
-        (c->maximized_vertical   ? CLIENT_MAXIMIZED_V : CLIENT_MAXIMIZED_NONE)|
-        (c->maximized_horizontal ? CLIENT_MAXIMIZED_H : CLIENT_MAXIMIZED_NONE)
+        (c->maximized_vertical   ? CLIENT_MAXIMIZED_V    : CLIENT_MAXIMIZED_NONE)|
+        (c->maximized_horizontal ? CLIENT_MAXIMIZED_H    : CLIENT_MAXIMIZED_NONE)|
+        (c->maximized            ? CLIENT_MAXIMIZED_BOTH : CLIENT_MAXIMIZED_NONE)
     );
-    const client_maximized_t next = s ? val : CLIENT_MAXIMIZED_NONE;
+    client_maximized_t next = s ? (val | current) : (current & (~val));
+
+    /* When both are already set during startup, assume `maximized` is true*/
+    if (next == (CLIENT_MAXIMIZED_H|CLIENT_MAXIMIZED_V) && !globalconf.loop)
+        next = CLIENT_MAXIMIZED_BOTH;
```

Comment:
> Shouldn't this by |= instead of =?

Nah, this is one of the possible "bad side effect" of this. If someone manually set both `maximized_[*]`, then it's interpreted as maximized on on restart. The opposite, if `=|` was used and the user had no shortcut for `maximized_[*]`, then he/she/it would be stuck with a triple maximized client with no way to turn it off. The "issue" itself already exist, but isn't "as bad" since the default maximized shortcut is currently "destructive" in the sense toggling it twice will reset both maximization. From now on, if one of the `maximize_*` is set and not the other, then it will be harder for the user to turn it off. The `=` just makes sure it doesn't happen because of this change.

At first I was thinking your idea of another maximized xprop was a good idea, but thinking about it again, I think a better course of action might be to reset fullscreen and maximization properties when restarting. It is a behavior change compared to previous version and someone, somewhere, will notice. However it simplifies the algorithm and users don't restart Awesome as often as they used to, so it wont be as noticeable as before. Anyway, when you lose your layouts and stack order, keeping maximization cause a confusing result.